src/libs/histogram.c

```diff
@@ -1184,7 +1184,8 @@ static void _lib_histogram_draw_vectorscope(dt_lib_histogram_t *d, cairo_t *cr,
     if(gtk_widget_get_visible(d->button_box))
     {
       // draw information about current selected harmony
-      char text[64];
+      char text[80]; // must be large enough to accommodate text translated into Cyrillic languages
```

Comment:
64 -> 80 is that safe enough? I mean future changes could break this again. I would probably go up to 128 to keep some room.

Reply:
For the current text, yes, it is enough with a margin (and the Ukrainian translation here is the longest of all Cyrillic ones). In order not to forget about this buffer when making changes, I intentionally left an explanatory comment. But if you think there is a need to make an even bigger margin, I will do it (I remember that the histogram module uses a lot of memory in the stack frame, but in the end a couple of tens of bytes should not be critical).

Reply:
the answer to this is to use `g_strdup_printf` (and `g_free`) instead of `snprintf`?

Reply:
@dterrahe I have already written about it, see above. And yes, of course we can use dynamic memory allocation at every opportunity. It is worth remembering that it has its price, but we can. Whether it is justified in each specific case needs to be analyzed and decided. But the main point is that my task here was to remove an obvious bug, and not to rewrite the original author's code in a different way. Therefore, the fix was made with minimal changes to the code.

As for your question, I don't know if this is the answer. Maybe that's the answer, maybe not. What does our coding style guide say about it? If this is indeed the answer, I will make a change to the PR.

Reply:
I would follow @dterrahe recommendation indeed, the `g_strdup_printf` will be ok for the foreseeable future.

Reply:
@TurboGit The main thing is to know which code will be accepted. 🙂
OK, I changed the implementation to use `g_strdup_printf`.